client/components/parts/SignUpForm.jsx

```diff
@@ -32,24 +33,41 @@ export default class SignUpForm extends PureComponent {
     hostUrl: 'uvic.club'
   }
 
+  handleSubmit = () => {
+
+    fetch(`http://www.hubsite.club/api/newsite/${this.state.siteInput}`, {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({'siteName': 'FIX ME'})
+    }).then(() => {
+      browserHistory.push(`/editor/${this.state.siteInput.toLowerCase()}`)
+    })
+  }
+
   siteChange = (e) => {
     const self = this
     const site = e.target.value
 
     this.setState({siteInput: site, siteInputState: 'testing'})
-
     this.validateSite(site, function(s){ self.setState({siteInputState: s}) })
   }
 
   validateSite = (site, callback) => {
-    clearTimeout(this.timer) // TODO: REMOVE once connect to server
+    clearTimeout(this.timer)
+    const self = this
     if (site === '') return callback('empty')
     else if (!/^[a-zA-Z0-9-]+$/.test(site)) return callback('invalid')
 
-    // TODO: Test with the DB, This simulates a server call
     this.timer = setTimeout(function() {
-      callback(site === 'taken' ? 'unavailable' : 'okay')
-    }, 200) // TODO: REMOVE once connect to server
+      Promise.resolve(fetch(`http://www.hubsite.club/api/site_exists/${self.state.siteInput}`, {
+        method: 'GET'
+      })).then((res) => {
+        res.text().then((t) => callback(t === 'true' ? 'unavailable' : 'okay'))
```

Comment:
Would be more readable if broken into separate lines, otherwise it's perfectly functional